### PR TITLE
instrumentation of asyncio tasks

### PR DESCRIPTION
**Pull request by arvidn**
This adds a developer tool that measures where tasks spend their wall-clock time. This can help pin-point delays that are not caused by needing to use a lot of CPU.

Here's an example output from syncing the wallet:
Task-33 dot

### CODE REVIEW

File: chia/server/start_wallet.py [hunk 1]
```diff
@@ -112,6 +114,8 @@ async def async_main() -> int:
 
 def main() -> int:
     freeze_support()
+    start_task_instrumentation()
+    atexit.register(stop_task_instrumentation)
```
Comment (arvidn) [inline chia/server/start_wallet.py:118]: this needs to be done conditionally, like with a command like flag. However, the wallet is started by the daemon, and it's not trivial to pass on that here. Any help would be appreciated

Comment (richardkiss) [reply chia/server/start_wallet.py:118]: A couple options I can think of:

- environment variables. I believe the daemon doesn't sanitize this, so stop the daemon, then do `CHIA_WALLET_INSTRUMENT=1 chia start wallet`
- magic file. Look for presence of `chia-wallet-instrument-magic` in the current directory. But I guess this presupposes that the `PWD` environment variable is passed through, so if this works, the first way would work

Comment (richardkiss) [reply chia/server/start_wallet.py:118]: Stop the daemon with `chia stop -d all`

Comment (mariano54) [reply chia/server/start_wallet.py:118]: Config option?

Comment (arvidn) [reply chia/server/start_wallet.py]: It has to happen before we start the asyncio run loop, and we load the config after that. I liked the idea of using an environment variable

**Review by mariano54 — APPROVED** [review]

**Pull request merged**